sunpy/map/mapbase.py

```diff
@@ -1482,6 +1484,10 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
         pad_x = int(np.max((diff[1], 0)))
         pad_y = int(np.max((diff[0], 0)))
 
+        if issubclass(self.data.dtype.type, numbers.Integral):
```

Comment:
Do we not need to check the missing value as well?

Reply:
yeah, it slipped my mind. I shall make a commit soon fixing all this.